groupsforsections.py

```diff
@@ -2,69 +2,152 @@
 import sys
 import logging
 import requests
-import urllib
 import utils
+from canvasapis import GroupsForSections
 import yaml
 from os.path import basename
 
-AUTHORIZATION = 'Authorization'
-BEARER = 'Bearer '
-MIME_TYPE_JSON = 'application/json'
-CONTENT_TYPE = 'Content-type'
 CONST_CANVAS = 'canvas'
 CONST_COURSE = 'course'
 CONST_TOKEN = 'token'
 CONST_URL = 'url'
 CONST_ID = 'id'
 CONST_GRP_CAT_NAME = 'group_category_name'
-HTTP_METHOD_POST = 'Post'
-HTTP_METHOD_GET = 'Get'
 
 
-def create_group_category():
-    logging.debug('create_group_category() Called')
+def create_group_category(group_category_name, groups_for_section, course_id):
+    """
+    The call creating the group category in a course, parse the Json, exception handling
+    :param String group_category_name:
+    :param GroupsForSections groups_for_section:
+    :return: String group category id
+    """
+    try:
+        response = groups_for_section.create_group_category(group_category_name, course_id)
 
-    url = canvas_url + '/api/v1/courses/' + course_id + '/group_categories?name=' + urllib.quote_plus(
-            group_category_name)
-    response = api_handler(url, HTTP_METHOD_POST)
-    if response is None:
-        logging.error('Api call for \'creating group categories\' is not successful')
+    except requests.exceptions.RequestException as e:
+        logging.exception('creating a group categories has erroneous response ' + e.message)
         sys.exit(1)
```

Comment:
Functions and methods do not feel like the right place to call `sys.exit()`.  Let the main program do it instead.  Then it will have the opportunity to log any final messages before exiting.

The functions and methods should be returning a value that indicates an error or throwing exceptions.